### PR TITLE
add code tabs in num9

### PR DESCRIPTION
**Pull request by benluo**

### CODE REVIEW

File: _overviews/scala3-book/taste-modeling.md [hunk 1]
```diff
@@ -50,27 +67,57 @@ trait Runner:
...
 ```
+{% endtab %}
+{% endtabs %}
 
 Given those traits, here’s a `Dog` class that extends all of those traits while providing a behavior for the abstract `speak` method:
 
+{% tabs traits-class class=tabs-scala-version %}
+{% tab 'Scala 2' for=traits-class %}
+```scala
+class Dog(name: String) extends Speaker, TailWagger, Runner {
```
Comment (bishabosha) [inline _overviews/scala3-book/taste-modeling.md:78]:
```suggestion
class Dog(name: String) extends Speaker with TailWagger with Runner {
```

File: _overviews/scala3-book/taste-modeling.md [hunk 2]
```diff
@@ -50,27 +67,57 @@ trait Runner:
...
+{% endtabs %}
 
 Notice how the class extends the traits with the `extends` keyword.
 
 Similarly, here’s a `Cat` class that implements those same traits while also overriding two of the concrete methods it inherits:
 
+{% tabs traits-override class=tabs-scala-version %}
+{% tab 'Scala 2' for=traits-override %}
+```scala
+class Cat(name: String) extends Speaker, TailWagger, Runner {
```
Comment (bishabosha) [inline _overviews/scala3-book/taste-modeling.md:99]:
```suggestion
class Cat(name: String) extends Speaker with TailWagger with Runner {
```

File: _overviews/scala3-book/taste-modeling.md [hunk 3]
```diff
@@ -50,27 +67,57 @@ trait Runner:
...
   override def startRunning(): Unit = println("Yeah ... I don’t run")
   override def stopRunning(): Unit = println("No need to stop")
 ```
+{% endtab %}
+{% endtabs %}
 
 These examples show how those classes are used:
 
+{% tabs traits-use class=tabs-scala-version %}
+{% tab 'Scala 2 and 3' for=traits-use %}
```
Comment (bishabosha) [inline _overviews/scala3-book/taste-modeling.md:120]: Cat and Dog are not case classes, so this syntax is not valid for Scala 2

File: _overviews/scala3-book/taste-modeling.md [hunk 4]
```diff
@@ -80,16 +127,32 @@ println(c.speak())      // "Meow"
...
 Here’s an example of a class that models a “person.” In OOP fields are typically mutable, so `firstName` and `lastName` are both declared as `var` parameters:
 
+{% tabs class_1 class=tabs-scala-version %}
+{% tab 'Scala 2' for=class_1 %}
+```scala
+class Person(var firstName: String, var lastName: String) {
+  def printFullName() = println(s"$firstName $lastName")
+}
+
+val p = Person("John", "Stephens")
```
Comment (bishabosha) [inline _overviews/scala3-book/taste-modeling.md:148]:
```suggestion
val p = new Person("John", "Stephens")
```

File: _overviews/scala3-book/taste-modeling.md [hunk 5]
```diff
@@ -99,17 +162,22 @@ println(p.firstName)   // "John"
...
+{% endtab %}
+{% endtabs %}
 
 Notice that the class declaration creates a constructor:
 
+{% tabs class_2 class=tabs-scala-version %}
+{% tab 'Scala 2 and 3' for=class_2 %}
 ```scala
 // this code uses that constructor
 val p = Person("John", "Stephens")
```
Comment (bishabosha) [inline _overviews/scala3-book/taste-modeling.md:174]: this is not valid in scala 2 without `new`

Comment (benluo) [reply _overviews/scala3-book/taste-modeling.md:216]: changed.

File: _overviews/scala3-book/taste-modeling.md (hunk 3 above)
Comment (benluo) [reply _overviews/scala3-book/taste-modeling.md]: changed.

**Pull request merged**